components/cdc/src/service.rs

```diff
@@ -142,21 +143,35 @@ impl ChangeData for Service {
                 })
         });
 
-        let rx = SizedBatchReceiver::new(rx, CDC_MAX_RESP_SIZE, Vec::new, |v, e| v.push(e));
-        let send_resp = sink.send_all(rx.then(|events| match events {
-            Ok(events) => {
-                let mut resp = ChangeDataEvent::default();
-                resp.set_events(events.into());
-                Ok((resp, WriteFlags::default()))
-            }
-            Err(e) => {
-                error!("cdc send failed"; "error" => ?e);
-                Err(Error::RpcFailure(RpcStatus::new(
-                    RpcStatusCode::UNKNOWN,
+        let rx = BatchReceiver::new(rx, CDC_MSG_MAX_BATCH_SIZE, Vec::new, |v, e| v.push(e));
+        let send_resp = sink.send_all(
+            rx.map(|events| {
+                let events_len = events.len();
+                let mut event_vecs = vec![Vec::with_capacity(events_len)];
```

Comment:
```suggestion
                let mut event_vecs = Vec::with_capacity(events_len);
```

Can we remove `stream::iter_ok` and `flatten`?